src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/Harness/Listeners.cs

```diff
@@ -0,0 +1,433 @@
+            }
+            else
+                throw new NotImplementedException();
+            Thread.Sleep(200);          // Calls are async, give them time to work.  
+        }
+        public override void Dispose()
+        {
+            m_session.Flush();
+            Thread.Sleep(1010);      // Let it drain.  TODO not clear this is needed.  
+            m_session.Dispose();     // This also will kill the real time thread 
+
+            if (m_dataFileName != null)
+            {
+                using (var traceEventSource = new ETWTraceEventSource(m_dataFileName))
+                {
+                    Console.WriteLine("Processing data file " + Path.GetFullPath(m_dataFileName));
+
+                    // Parse all the events as as best we can, and also send unhandled events there as well.  
+                    traceEventSource.Registered.All += OnEventHelper;
+                    traceEventSource.Dynamic.All += OnEventHelper;
+                    traceEventSource.UnhandledEvents += OnEventHelper;
+                    // Process all the events in the file.  
+                    traceEventSource.Process();
+                    Console.WriteLine("Done processing data file " + Path.GetFullPath(m_dataFileName));
+                }
+            }
+        }
+
+#region private
+        void OnEventHelper(TraceEvent data)
+        {
+            // Ignore manifest events. 
+            if ((int)data.ID == 0xFFFE)
+                return;
+            this.OnEvent(new EtwEvent(data));
+        }
+
+        /// <summary>
+        /// EtwEvent implements the 'Event' abstraction for ETW events (it has a TraceEvent in it) 
+        /// </summary>
+        internal class EtwEvent : Event
+        {
+            public override bool IsEtw { get { return true; } }
+            public override string ProviderName { get { return m_data.ProviderName; } }
+            public override string EventName { get { return m_data.EventName; } }
+            public override object PayloadValue(int propertyIndex, string propertyName)
+            {
+                if (propertyName != null)
+                    Assert.Equal(propertyName, m_data.PayloadNames[propertyIndex]);
+                return m_data.PayloadValue(propertyIndex);
+            }
+            public override string PayloadString(int propertyIndex, string propertyName)
+            {
+                Assert.Equal(propertyName, m_data.PayloadNames[propertyIndex]);
+                return m_data.PayloadString(propertyIndex);
+            }
+            public override int PayloadCount { get { return m_data.PayloadNames.Length; } }
+            public override IEnumerable<string> PayloadNames { get { return m_data.PayloadNames; }}
+
+#region private
+            internal EtwEvent(TraceEvent data) { m_data = data.Clone(); }
+
+            TraceEvent m_data;
+#endregion
+        }
+
+        string m_dataFileName;
+        volatile TraceEventSession m_session;
```

Comment:
Can you run the codeformatter tool over this code?  There are certain style things that should be addressed across the code, e.g. fields like these should be explicitly declared private, they should begin with ```_``` rather than ```m_```, etc.  The codeformatter tool should take care of most of those.